src/main/java/Sam.java

```diff
@@ -0,0 +1,129 @@
+                    break;
+                case "mark":
+                    // Marking a task as done
+                    listIndex = Integer.parseInt(words[1]) - 1;
+                    markCheck(listIndex, numItems);
+                    records[listIndex].isDone = true;
+                    System.out.println("Nice! I've marked this task as done:\n" + records[listIndex]);
+                    break;
+                case "unmark":
+                    // Marking a task as not done
+                    listIndex = Integer.parseInt(words[1]) - 1;
+                    markCheck(listIndex, numItems);
+                    records[listIndex].isDone = false;
+                    System.out.println("OK, I've marked this task as not done yet:\n" + records[listIndex]);
+                    throw new InvalidInputException();
+                    //break;
+                case "todo":
+                    // Adding a new todo task
+                    String todoDescription = line.substring(5); // Extracting description
+                    records[numItems] = new Todo(todoDescription);
+                    System.out.println("Got it. I've added this task:\n" + records[numItems]);
+                    numItems++;
+                    break;
+                case "deadline":
+                    // Adding a new deadline task
+                    String deadlineDescription = line.substring(9); // Extracting description
+                    String[] deadlineParts = deadlineDescription.split(" /by ");
+                    records[numItems] = new Deadline(deadlineParts[0], deadlineParts[1]);
+                    System.out.println("Got it. I've added this task:\n" + records[numItems]);
+                    numItems++;
+                    break;
+
+                case "event":
+                    // Adding a new event task
+                    String eventDescription = line.substring(6); // Extracting description
+                    String[] eventParts = eventDescription.split(" /from | /to ");
+                    records[numItems] = new Event(eventParts[0], eventParts[1], eventParts[2]);
+                    System.out.println("Got it. I've added this task:\n" + records[numItems]);
+                    numItems++;
+                    break;
+                default:
+                    // Invalid command
+                    System.out.println("No valid command detected, please try again.");
+                }
+                printLine();
+                line = in.nextLine();
+            } catch (InvalidInputException e) {
+                System.out.println("Try again!");
+                printLine();
+                line = in.nextLine(); // Prompt for new input
+            } catch (NumberFormatException e) {
+                System.out.println("That's not a number!");
+                printLine();
+                line = in.nextLine(); // Prompt for new input
+            } catch(IndexOutOfBoundsException e){
```

Comment:
There needs to be a space between 'catch' and (, and, ) and {, as per coding standards